### DISCUSSION TITLE
Error while running with SEMFEM preconditioner

### DISCUSSION THREAD

**Discussion by nandu90** [Nek5000/nekRS#503] [Compiling]
I am getting the following error while running with SEMFEM preconditioner, using v23.0.0 (4f87e0e)

ERROR: Recompile with HYPRE GPU support!

This is on the nek5k machine, NVIDIA-A100. The code runs fine with the default (semg) pre-conditioner on GPUs

![image](https://github.com/Nek5000/nekRS/assets/26067602/7e7c60dc-725d-4918-927d-465169e808ea)

Seems like I am missing something during compilation. Any ideas?

Thanks!

**Comment by stgeke** [comment]
You want to compile with `-DENABLE_AMGX=ON` or `-DENABLE_HYPRE_GPU=ON`. On systems with NVIDIA GPUs I would recommend to use `AMGX` at least for now.

**Reply by nandu90** [reply]
Thanks @stgeke 

I tried with either of the options on Polaris. Get the following error:

![image](https://github.com/Nek5000/nekRS/assets/26067602/bf2523af-5b52-4769-a509-f985eb65aee1)

On nek5k the code quits without throwing a helpful message.
